presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/ResourceGroup.java

```diff
@@ -25,37 +25,68 @@
 
     DataSize getSoftMemoryLimit();
 
+    /**
+     * Threshold on total distributed memory usage after which new queries
+     * will queue instead of starting.
+     */
     void setSoftMemoryLimit(DataSize limit);
 
     Duration getSoftCpuLimit();
 
+    /**
+     * Threshold on total distributed CPU usage after which max running queries will be reduced.
+     */
     void setSoftCpuLimit(Duration limit);
 
     Duration getHardCpuLimit();
 
+    /**
+     * Threshold on total distributed CPU usage after which new queries
```

Comment:
It's a bit confusing that the queueing is controlled by the "hard" cpu limit vs the "soft" memory limit